File: app/repositories/health.py
from sqlalchemy.orm import Session

from app.db.models.health import HealthStatus


def save_health_status(db: Session, status: str):
    """
    Save the health status to the database.

    Args:
        db (Session): The database session.
        status (str): The status string to save (e.g., "ok").
    """
    print(f"Saving health status: {status}")
    health = HealthStatus(status=status)
    db.add(health)
    db.commit()

File: rust-app/src/repositories/health.rs
use sqlx::PgPool;
use tracing::info;

/// Save the health status to the database.
///
/// Mirrors Python's save_health_status from app/repositories/health.py.
///
/// # Arguments
/// * `pool` - The database connection pool.
/// * `status` - The status string to save (e.g., "ok").
pub async fn save_health_status(pool: &PgPool, status: &str) -> Result<(), sqlx::Error> {
    info!("Saving health status: {}", status);
    sqlx::query("INSERT INTO health_status (status, timestamp) VALUES ($1, NOW())")
        .bind(status)
        .execute(pool)
        .await?;
    Ok(())
}
